src/main/java/be/syntra/devshop/DevshopBack/services/SearchServiceImpl.java

```diff
@@ -22,61 +30,116 @@ public SearchServiceImpl(ProductService productService) {
     }
 
     @Override
-    public List<Product> applySearchModel(SearchModel searchModel) {
-        List<Product> productList;
-
-        productList = (searchModel.isArchivedView())
-                ? productService.findAllByArchivedTrue()
-                : productService.findAllByArchivedFalse();
-
-        if (StringUtils.hasText((searchModel.getSearchRequest()))) {
-            productList = productList.parallelStream()
-                    .filter(product -> product.getName()
-                            .toLowerCase()
-                            .contains(searchModel.getSearchRequest().toLowerCase()))
-                    .collect(Collectors.toUnmodifiableList());
+    public ProductPageAndMinMaxPrice applySearchModel(SearchModel searchModel) {
+        log.info("searchModel -> {}", searchModel);
+        Pageable pageable = getSortingPageable(setDefaultPaginationValues(searchModel));
+        if (searchModel.isSearchResultView()) {
+            if (StringUtils.hasText(searchModel.getSearchRequest())) {
+                if (isPriceFilterActiveAndValid(searchModel)) {
+                    return getSearchResultsAndMinMaxPrice(
+                            productService.findAllNonArchivedBySearchTermAndPriceBetween(
+                                    searchModel.getSearchRequest(), searchModel.getPriceLow(), searchModel.getPriceHigh(), pageable),
+                            productService.findMinPriceProductNonArchivedBySearchTermAndPriceBetween(
```

Comment:
tl;dr -> referring to Slack for discussion about min-max price

Reply:
may i suggest to revisit this when we're sure the behavior of searches is ok? I want to be sure to not go into this with a wrecking ball when it might be still not as you implemented the original search

Reply:
flow is restored to original search. We'll take this to another ticket